tests/unit/test.config.js

```diff
@@ -456,14 +508,244 @@ describe('config', () => {
+          configObject});
+      }, UsageError, 'UsageError: The config file at some/path/to/config.js ' +
+        'specified the type of "apiUrl" incorrectly');
+    });
+
+    it('throws an error when the type of one of option values' +
+      ' is invalid', () => {
+      const params = makeArgv({
+        userCmd: ['sign'],
+        command: 'sign',
+        commandOpt: {
+          'api-url': {
+            requiresArg: true,
+            type: 'string',
+            demand: false,
+            default: 'pretend-default-value-of-apiKey',
+          },
+          'api-key': {
+            requiresArg: true,
+            type: 'string',
+            demand: false,
+            default: 'pretend-default-value-of-apiKey',
+          },
+        },
+      });
+      const configObject = {
+        sign: {
+          apiUrl: 2,
+          apiKey: 'fake-api-key',
+        },
+      };
+      assert.throws(() => {
+        applyConf({...params,
+          configObject});
+      }, UsageError, 'UsageError: The config file at some/path/to/config.js ' +
+        'specified the type of "apiUrl" incorrectly');
+    });
+
+    it('throws an error when the type in option config is missing', () => {
+      const params = makeArgv({
+        userCmd: ['sign'],
+        command: 'sign',
+        commandOpt: {
+          'api-url': {
+            requiresArg: true,
+            demand: false,
+            default: 'pretend-default-value-of-apiKey',
+          },
+        },
+      });
+      const configObject = {
+        sign: {
+          apiUrl: 2,
+          apiKey: 'fake-api-key',
+        },
+      };
+      assert.throws(() => {
+        applyConf({...params,
+          configObject});
+      }, UsageError,
+        'UsageError: Option: apiUrl was defined without a type.');
+    });
+
+    it('throws an error when the type of one of them is in config' +
+      ' missing', () => {
+      const params = makeArgv({
+        userCmd: ['sign'],
+        command: 'sign',
+        commandOpt: {
+          'api-url': {
+            requiresArg: true,
+            demand: false,
+            default: 'pretend-default-value-of-apiKey',
+          },
+          'api-key': {
+            requiresArg: true,
+            demand: false,
+            type: 'string',
+            default: 'pretend-default-value-of-apiKey',
+          },
+        },
+      });
+      const configObject = {
+        sign: {
+          apiUrl: 2,
+          apiKey: 'fake-api-key',
+        },
+      };
+      assert.throws(() => {
+        applyConf({...params,
+          configObject});
+      }, UsageError,
+        'UsageError: Option: apiUrl was defined without a type.');
+    });
+
+    it('throws an error when the type of an unrelated sub option is invalid',
+    () => {
+      const program = new Program(['run']);
+
+      const commandDesc = 'this is a fake command';
+      const commandExecutor = sinon.stub();
+
+      program.command('run', commandDesc, commandExecutor,
```

Comment:
You can replace `commandExecutor` here and below with `sinon.stub()`; it's shorter ;)